pkg/pprof/pprof.go

```diff
@@ -1195,3 +1175,135 @@ func Unmarshal(data []byte, p *profilev1.Profile) error {
 	}
 	return p.UnmarshalVT(buf.Bytes())
 }
+
+func sanitizeReferences(p *profilev1.Profile) {
+	ms := int64(len(p.StringTable))
+	// Handle the case when "" is not present,
+	// or is not at string_table[0].
+	z := int64(-1)
+	for i, s := range p.StringTable {
+		if s == "" {
+			z = int64(i)
+		}
+	}
+	if z == -1 {
+		// No empty string found in the table.
+		// Reduce number of invariants by adding one.
+		z = ms
+		p.StringTable = append(p.StringTable, "")
+		ms++
+	}
+	// Swap zero string.
+	p.StringTable[z], p.StringTable[0] = p.StringTable[0], p.StringTable[z]
+	// Now we need to update references to strings:
+	// invalid references (>= len(string_table)) are set to 0.
+	// references to empty string are set to 0.
+	str := func(i int64) int64 {
+		if i == 0 && z > 0 {
+			// z > 0 indicates that "" is not at string_table[0].
+			// This means that element that used to be at 0 has
+			// been moved to z.
+			return z
+		}
+		if i == z || i >= ms {
```

Comment:
We probably should also cover this case: 

```suggestion
		if i == z || i >= ms || i < 0 {
```